docs/auth/oidc.md

```diff
@@ -0,0 +1,234 @@
+---
+id: oidc
+title: OIDC provider from scratch
+description: This section shows how to use an OIDC provider from scrath, same steps apply for custom providers.
+---
+
+This section shows how to use an OIDC provider from scratch, same steps apply for custom
+providers. Please note these steps are for using a provider, not how to implement one.
+
+## Summary
+
+To add providers not enabled by default like OIDC, we need to follow some steps, we
+assume you already have a sign in page to which we'll add the provider so users can
+sign in through the provider. In simple steps here's how you enable the provider:
+
+- Create an API reference to identify the provider.
+- Create the API factory that will handle the authentication.
+- Add a resolver so you can handle the result from the authentication.
+- Configure the provider to access your 3rd party auth solution.
+- Add the provider to sign in page so users can login with it.
+
+We'll explain each step more in detail next.
+
+### The API reference
+
+An API reference exist for the sake of **Dependency Injection**, check [Utility APIs][4]
+for extended explanation.
+
+In this OIDC example, we'll create the API reference directly in the
+`packages/app/src/apis.ts` file, it is not a requirement to put the reference in this
+file. Any location will do as long as it's available to be imported to where the API
+factory is, as well as easily accessible to the rest of the application so any package
+and plugin can inject the API instance when necessary.
+
+An example of such would be when you use an auth provider from a library installed with
+NPM, or any other library repository, you would import the API ref from the library.
+
+```ts
+export const oidcAuthApiRef: ApiRef<
+  OpenIdConnectApi & ProfileInfoApi & BackstageIdentityApi & SessionApi
+> = createApiRef({
+  id: 'core.auth.oidc',
+});
```

Comment:
I'm thinking we should encourage this ref to be named according to the actual identity provider. For example, if we use the generic provider to auth towards "Acme Org", we might call this `acmeAuthApiRef`. Likewise we'd give it an ID like `'acme.auth'`.

Reply:
We don't know which tokenProvider the user will set in the config yaml from the default supported ones so I think using OIDC here is correct.

Reply:
Changed to a custom name, in this example for `core.auth.azureOIDC`.